src/pineko/kfactor.py

```diff
@@ -209,148 +193,108 @@ def create_singlegridonly(
     return new_grid
 
 
-def create_grids(
-    gridpath,
-    max_as,
-    first_non_zero_as_order,
-    min_al,
-    centrals_k_factor,
+def do_it(
+    central_kfactor,
     alphas,
+    grid,
+    pto_to_update,
+    target_grid_path,
     order_exists,
 ):
-    """Create all the necessary grids for a certain starting grid."""
-    grid = pineappl.grid.Grid.read(gridpath)
-    m_value = first_non_zero_as_order
-    nec_orders = compute_orders_map(m_value, max_as, min_al, order_exists)
-    grid_list = {}
-    for to_construct_order in nec_orders:
-        list_grid_order = []
-        for nec_order in nec_orders[to_construct_order]:
-            list_grid_order.append(
-                create_singlegridonly(
-                    grid,
-                    m_value,
-                    nec_order,
-                    to_construct_order,
-                    centrals_k_factor,
-                    alphas,
-                    order_exists,
-                )
-            )
-        grid_list[to_construct_order] = list_grid_order
+    """Apply the centrals_kfactor to the grid.
 
-    return grid_list, nec_orders
-
-
-def is_already_in(to_check, list_orders):
-    """Check if the requested order is already in the grid."""
-    for order in list_orders:
-        if (
-            order[-2] == 0
-            and order[-1] == 0
-            and (order[0] == to_check[0])
-            and (order[1] == to_check[1])
-        ):
-            return True
-    return False
+    Parameters
+    ----------
+    central_kfactor : np.ndarray
+        kfactors to apply
+    alphas : lhapdf.AlphaS
+        alphas
+    grid : pineappl.grid
+        loaded grid
+    pto_to_update : int
+        perturbative order to update: 1 = LO, 2 = NLO ...
```

Comment:
Wait a second this is not NNPDF default: LO <=> PTO=0

Reply:
Indeed that is why I started this PR 🙃

Reply:
Indeed this is not NNPDF conventions for which PTO=2 is NNLO, for example. However this is EKO convention and thus, in general, pineline convention. Since `pineko` is part of the pineline it seemed sensible to me to use the `pineline` convention, even if here we are using NNPDF kfactors (note in fact that we are not limited to NNPDF kfactor, but we can use any kfactor written in the same format).

Reply:
:scream: I mean a name is just a name, but then let's NOT use the letter combination `pto` and let's call it `order_to_update` e.g. else it is really confusing

Reply:
well, you're more experienced than myself here, so do as you feel more comfortable. However, please add a fool-proof documentation somewhere! e.g. add this discussion together with a big warning in the docstring

Reply:
done in [21572f8](https://github.com/NNPDF/pineko/pull/161/commits/21572f8492360fe951eb0920810fb1a218430d41)